frontend/internal-packages/db/supabase/migrations/20250814082520_create_public_share_realtime.sql

```diff
@@ -0,0 +1,118 @@
+-- Table to track which design sessions are publicly shared
+create table if not exists "public"."public_share_settings" (
+    "design_session_id" "uuid" not null,
+    "created_at" timestamp with time zone default "now"() not null
+);
+alter table "public"."public_share_settings" owner to "postgres";
+
+-- Public views for anonymous access
+-- These views expose limited columns to prevent unintended data exposure
+create view public.artifacts_public as
+select
+    id,
+    design_session_id,
+    artifact,
+    created_at,
+    updated_at
+from public.artifacts;
+
+create view public.building_schema_versions_public as
+select
+    id,
+    building_schema_id,
+    number,
+    created_at,
+    patch,
+    reverse_patch
+from public.building_schema_versions;
+
+create view public.building_schemas_public as
+select
+    id,
+    design_session_id,
+    schema,
+    created_at,
+    git_sha,
+    initial_schema_snapshot,
+    schema_file_path
+from public.building_schemas;
+
+-- Limited columns for design sessions to hide sensitive organization data
+create view public.design_sessions_public as
+select
+    id,
+    name,
+    created_at,
+    parent_design_session_id
+from public.design_sessions;
+
+create view public.timeline_items_public as
+select
+    id,
+    design_session_id,
+    content,
+    created_at,
+    updated_at,
+    building_schema_version_id,
+    type,
+    query_result_id,
+    assistant_role
+from public.timeline_items;
+
+-- Grant read access to anonymous users for public views
+grant select on table "public"."public_share_settings" to "anon";
+grant select on table "public"."artifacts_public" to "anon";
+grant select on table "public"."building_schema_versions_public" to "anon";
+grant select on table "public"."building_schemas_public" to "anon";
+grant select on table "public"."design_sessions_public" to "anon";
+grant select on table "public"."timeline_items_public" to "anon";
```

Comment:
this five are views.  `grant select on table`(not  `grant select on view`) is needed.